src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -31,10 +31,14 @@ object TruthTable {
 
   /** Convert a table and default output into a [[TruthTable]]. */
   def apply(table: Iterable[(BitPat, BitPat)], default: BitPat, sort: Boolean = true): TruthTable = {
-    require(table.map(_._1.getWidth).toSet.size == 1, "input width not equal.")
+    val inputWidth = table.map(_._1.getWidth).max
     require(table.map(_._2.getWidth).toSet.size == 1, "output width not equal.")
     val outputWidth = table.map(_._2.getWidth).head
-    val mergedTable = table
+    val mergedTable = table.map {
+      case (in, out) =>
+        // pad input signals.
+        (BitPat.dontCare(inputWidth - in.getWidth) ## in, out)
+    }
```

Comment:
```suggestion
    val mergedTable = table.map {
      // pad input signals if necessary
      case (in, out) if inputWidth > in.width =>
        (BitPat.dontCare(inputWidth - in.width) ## in, out)
      case (in, out) => (in, out)
    }
```
I'm not sure that there's any negative to unconditionally padding but it might be worth only padding if necessary? Any thoughts?

I also switched it to use `.width` instead of `.getWidth` because the latter reminds me of `Data.getWidth` which can fail. `.width` is more clear that this is always known (for BitPats).